omegaconf/basecontainer.py

```diff
@@ -52,14 +52,15 @@ def _resolve_with_default(
         """returns the value with the specified key, like obj.key and obj['key']"""
 
         def is_mandatory_missing(val: Any) -> bool:
-            return get_value_kind(val) == ValueKind.MANDATORY_MISSING  # type: ignore
+            return bool(get_value_kind(val) == ValueKind.MANDATORY_MISSING)
 
-        value = _get_value(value)
+        val = _get_value(value)
         has_default = default_value is not DEFAULT_VALUE_MARKER
-        if has_default and (value is None or is_mandatory_missing(value)):
+        if has_default and (val is None or is_mandatory_missing(val)):
             return default_value
 
         resolved = self._resolve_interpolation(
+            parent=self,
             key=key,
             value=value,
             throw_on_missing=not has_default,
```

Comment:
This looks like an actual change.
Logic before was passing the output of _get_value() to _resolve_interpolation().
_get_value() unwrap ValueNode() which means you may be passing ValueNode ti _resolve_interpolation now. is this intentional?

Reply:
Yes this is intentional. IIRC, the motivation is for string interpolations like `"abc_${foo}"` where we need to create a new `StringNode` and obtain the `optional` flag from the original node's metadata, but if we only have access to the value then this information is lost.